homeassistant/components/homewizard/update.py

```diff
@@ -0,0 +1,85 @@
+"""HomeWizard platform that offers update entities."""
+from __future__ import annotations
+
+from homewizard_energy.const import LATEST_STABLE_FIRMWARE
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+from .entity import HomeWizardEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update platform."""
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            HomeWizardUpdateEntity(
+                coordinator,
+                entry,
+            ),
+        ]
+    )
+
+
+class HomeWizardUpdateEntity(HomeWizardEntity, UpdateEntity):
+    """Representation of a HomeWizard update entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+    _attr_name = "Firmware"
```

Comment:
```suggestion
```
This can be omitted as Update entities get their (translated) name from the device class.